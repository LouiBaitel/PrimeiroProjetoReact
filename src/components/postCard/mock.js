export const postCardPropsMock = {
    title: 'title one',
    body: 'body one',
    id: 1,
    cover:'img/img.png'
}