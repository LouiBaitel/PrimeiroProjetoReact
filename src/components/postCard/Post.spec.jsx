import { render, screen } from '@testing-library/react'
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', ()=>{
    it('should render PosCard correctly', ()=>{
        //Renderiza o componente que est√° sendo testado
        //const {debug} = render(<PostCard {...props} />);
        //debug();

        render(<PostCard {...props} />);

        expect(screen.getByRole('img', {name: /title one/i}))
            .toBeInTheDocument('src', 'img/img.png');

        expect(screen.getByRole('heading', {name: /title one/i})).toBeInTheDocument();

        expect(screen.getByText('body one')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const {container} = render(<PostCard {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});