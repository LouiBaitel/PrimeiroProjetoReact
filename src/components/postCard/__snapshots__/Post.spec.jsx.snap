// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title one"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
       
      title one
       
    </h1>
    <p>
       
      body one
       
    </p>
  </div>
</div>
`;
